compiler/AST/type.cpp

```diff
@@ -1166,6 +1172,99 @@ Symbol* AggregateType::getField(int i) {
   return toDefExpr(fields.get(i))->sym;
 }
 
+
+void AggregateType::printDocs(std::ostream *file, unsigned int tabs) {
+  if (this->symbol->hasFlag(FLAG_NO_DOC) || this->isUnion()) {
+    return;
+  }
+
+  this->printTabs(file, tabs);
+  *file << this->docsDirective();
+  *file << this->symbol->name;
+  *file << this->docsSuperClass();
+  *file << std::endl;
+
+  // In rst mode, ensure there is an empty line between the class/record
+  // signature and its description or the next directive.
+  if (!fDocsTextOnly) {
+    *file << std::endl;
+  }
+
+  if (this->doc != NULL) {
+    this->printDocsDescription(this->doc, file, tabs + 1);
+    *file << std::endl;
+
+    // In rst mode, ensure there is an empty line between the class/record
+    // description and the next directive.
+    if (!fDocsTextOnly) {
+      *file << std::endl;
+    }
+  }
+}
+
+
+/*
+ * Returns super class string for documentation. If super class exists, returns
+ * ": <super class name>".
+ */
+std::string AggregateType::docsSuperClass() {
+  if (this->inherits.length > 0) {
+    std::vector<std::string> superClassNames;
+
+    for_alist(expr, this->inherits) {
+      if (UnresolvedSymExpr* use = toUnresolvedSymExpr(expr)) {
+        superClassNames.push_back(use->unresolved);
+      }
+    }
+
+    if (superClassNames.empty()) {
+      return "";
+    }
+
+    // If there are super classes, join them into a single comma delimited
+    // string prefixed with a colon.
+    std::string superClasses = " : " + superClassNames.front();
+    for (unsigned int i = 1; i < superClassNames.size(); i++) {
+      superClasses += ", " + superClassNames.at(i);
+    }
+    return superClasses;
+  } else {
+    return "";
+  }
+}
+
+
+/*
+ * Add "inherited from <name>" to the documentation.
+ */
+void AggregateType::printInheritanceDocs(std::ostream *file, unsigned int tabs) {
```

Comment:
This method is not used by anything.  Do we want to comment it out until we've looked into the design implications?